x/incentives/types/incentive.go

```diff
@@ -0,0 +1,48 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+	ethermint "github.com/tharsis/ethermint/types"
+)
+
+// NewIncentive returns an instance of Incentive
+func NewIncentive(
+	contract common.Address,
+	allocations sdk.DecCoins,
+	epochs uint32,
+) Incentive {
+	return Incentive{
+		Contract:    contract.String(),
+		Allocations: allocations,
+		Epochs:      epochs,
+	}
+}
+
+// Validate performs a stateless validation of a Incentive
+func (i Incentive) Validate() error {
```

Comment:
we need to create one of these for the gas meter too

Reply:
added